woocommerce/woocommerce-gutenberg-products-block
# Remove "package" tag from files inside "src" and "tests/php"

On WooCommerce core we don't require a `@package` tag inside the `src` directory anymore, see: https://github.com/woocommerce/woocommerce/pull/27239
We are doing it because those classes already use namespaces and those will get rendered correctly by our code reference: https://docs.woocommerce.com/wc-apidocs/

Good news is that Blocks only use code with namespaces, so mostly of the time you folks don't need to worry about a `@package` tag.

--- Below Added by @nerrad  ----

I continued the changes @claudiosanches started in this pull. This pull includes:

- Adding exclusions to WooCommerce PHPCS rules for file comment block and tag requirements. This is not needed for namespace files.
- Remove file comment blocks for all namespaced files.
- Remove the unnecessary usage of `defined( 'ABSPATH' ) || exit;` from files. This is a holdover from days when files had executed code. With class declarations this is no longer the case.
- Implemented `@internal` tag on all `StoreAPI` classes to make explicit these are internal only. Also added this tag to a couple other classes that are internal only (I think that was Cart and Checkout blocks and Atomic block classes). While there's others that could probably use this, I think the ship has sailed on the public status of the classes. Even the ones I marked internal only we'll have to make changes on carefully.

## To Test

The impact of these changes is nil on deployed code because it's just phpdoc block changes. Any syntax mistakes (accidental removal of an import, or non-closing comment block) should surface fairly quickly in existing linting and automated tests).

So there should be no need for manual testing of this pull.
